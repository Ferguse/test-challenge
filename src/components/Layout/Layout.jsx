import { Outlet } from 'react-router-dom';
import { Header } from 'components/Header';
import { Footer } from 'components/Footer';
import { Nav } from 'components/Nav';

export const Layout = () => {
  return (
    <div>
      <Header />
      <Nav />
      <Outlet />
      <Footer />
    </div>
  );
};
