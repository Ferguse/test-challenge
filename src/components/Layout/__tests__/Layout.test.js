import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockedStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { Layout } from '../Layout';

describe('component [Layout]', () => {
  const state = {
    wishList: {},
    categories: {
      entities: [],
    },
    current: {},
  };

  const store = configureMockedStore([thunk])(state);
  const history = createMemoryHistory({ initialEntries: ['/home'] });

  it('Renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history} location={history.location} navigator={history}>
            <Layout />
          </Router>
        </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
