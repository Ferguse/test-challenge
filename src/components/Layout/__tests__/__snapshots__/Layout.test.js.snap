// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component [Layout] Renders 1`] = `
<div>
  <header
    className="header bg-white"
  >
    <div
      className="container flex justify-between"
    >
      <div
        className="bold logo"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          ELITE SPORT
        </a>
      </div>
      <div>
        <a
          href="/wish-list"
          onClick={[Function]}
        >
          <i
            className="fi fi-rs-heart"
          />
          <span>
             Wish list 
          </span>
        </a>
      </div>
    </div>
  </header>
  <nav
    className="nav"
  >
    <div
      className="container"
    >
      <ul
        className="nav-list flex"
      >
        <li
          className="nav-item"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <i
              className="fi fi-rs-home nav-icon"
            />
            <span>
               Home
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/category/bike"
            onClick={[Function]}
          >
            <i
              className="fi fi-rs-bike nav-icon"
            />
            <span>
               
              bike
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/category/hiking"
            onClick={[Function]}
          >
            <i
              className="fi fi-rs-hiking nav-icon"
            />
            <span>
               
              hiking
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/category/skating"
            onClick={[Function]}
          >
            <i
              className="fi fi-rs-skating nav-icon"
            />
            <span>
               
              skating
            </span>
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <footer>
    <div
      className="container"
    >
      <p>
        Ipsum.
      </p>
      <p>
        Dolor nostrud est cupidatat aute magna cupidatat do consequat magna nisi voluptate. Sit aliquip exercitation nulla proident sint sint amet exercitation mollit anim aliquip velit. Do adipisicing Lorem laboris enim velit sit nulla sint nisi excepteur consequat id ad exercitation. Irure labore incididunt.
      </p>
      <p>
        Et ad et incididunt tempor.
      </p>
    </div>
  </footer>
</div>
`;
